internal/sharedcomponent/sharedcomponent.go

```diff
@@ -95,17 +90,23 @@ func (r *SharedComponent[V]) Start(ctx context.Context, host component.Host) err
 			_ = r.telemetry.ReportComponentStatus(component.NewPermanentErrorEvent(err))
 		}
 	})
+	r.activeCount.Add(1)
```

Comment:
Implementing this is a problem, as I mentioned in the comment in the other PR. The problem is that we have a contract that says after calling shutdown on a component that component cannot emit to the telemetry pipeline anymore. This will not be the case for example for the otel receiver because if I have a Traces and Metrics receiver that will share this when I shutdown the first one it will still emit telemetry.

So I am not sure how we can do this, we may have to stick with the current implementation which is to shutdown on the first request.

Reply:
understood ; I can add tests to check this contract is enforced well.

Reply:
We have something https://github.com/open-telemetry/opentelemetry-collector/blob/main/receiver/receivertest/contract_checker.go but does not check for the scenario I mentioned: Receiver supports multiple signal types, start an instance of multiple types on same port (like otlp receiver) then shutdowns one of them and expects to not add any data to the pipeline for that signal.

Reply:
I may have a way to go about this, but I'm going to break this changeset in 2, one is going to be just the API changes, and the other the changes in the lifecycle. I have too much going on in this PR.